### ISSUE TITLE
Internationalization/multi language

### ISSUE THREAD

**Issue by mariuskagc** [MISP/MISP#596]
Hello.

I've checked that the multi language functionality is not being used right now, the files in the LOCATE folder are empty and all the text is hardcoded in the HTMLs.

Is there any plan to change this behaviour soon (to support more languages I mean)?

Thank you in advance!!

**Comment by cvandeplas** [comment]
Currently there was no real demand from the community for localisation. 

Adding it is indeed not too much work as CakePHP supports it out of the box.  (http://book.cakephp.org/2.0/en/core-libraries/internationalization-and-localization.html) 

However you are correct that the current code has hardcoded text in many many places.
We could change it over the whole codebase at once, however this will cost a few days of full-time work. 
In short, all the user-visible strings from the tpl files should be replaced with <?php echo __("text") ; ?>. There might be a few minor things in the models and behaviours though. 

The step afterwards is translating everything.

What languages do you have in mind?

**Comment by mariuskagc** [comment]
Hi again.

We would like to have Spanish language available, as it is the second language in the world (Ref https://en.wikipedia.org/wiki/List_of_languages_by_number_of_native_speakers).
If you need it we could help you with the translation  :)

**Comment by SteveClement** [comment]
Dear @mariuskagc in case you are still a MISP User or know others who are, we are happy to announce that the community can now contribute their Internationalization efforts to MISP :)

Here you see the PO Templates: https://github.com/MISP/MISP/tree/2.4/app/Locale
The most widely used PO editor seems to be this one: https://poedit.net/

A first read would be this:

https://book.cakephp.org/2.0/en/core-libraries/internationalization-and-localization.html

More details about translations under the hood: https://book.cakephp.org/2.0/en/core-libraries/behaviors/translate.html (FYI only)

For the sake of completeness here is more documentation about the cake i18n shell: https://book.cakephp.org/2.0/en/console-and-shells/i18n-shell.html

Important: Please add a translation file to the misp-book repository.
Here are some instructions: https://github.com/MISP/misp-book/tree/master/translation

Thanks for your future contribution.
